elastic/kibana
# [Guided onboarding] Observability tour bug fixes

Fixes https://github.com/elastic/kibana/issues/135927

This PR addresses 2 issues in the Observability tour that were brought up on https://github.com/elastic/kibana/pull/135095#pullrequestreview-1026936369.

### How to test
**Scenario 1:** Guided onboarding button click closes tour

1. Load the sample logs data.
2. Clear localStorage
3. Navigate to `app/home/#/getting_started`
4. Click "Monitor my infrastructure".
5. Complete the tour up to the last step - "Now add your data!" **(Do not click "End tour")**
6. Click on the "Guided setup" button. Verify the tour closes.

https://user-images.githubusercontent.com/5226211/179023155-88d399f1-5193-48e6-a70c-c29189aa4af2.mp4

**Scenario 2:** Clicking on a link highlighted by the tour does not dismiss the tour

1. This issue was originally discovered against edge-oblt data via CCS (`oblt-cli cluster create ccs --cluster-name=edge-oblt`). However, you could also reproduce by having a fresh instance with no logs data.
2. Clear localStorage
3. Navigate to `app/home/#/getting_started`
4. Click "Monitor my infrastructure".
5. Go to the 2nd step in the tour - "Tail your infrastructure logs in real time"
6. Click on the "Streams" link.
7. Verify the tour still displays on the "Streams" page.

The code was incorrectly checking that both the data was loaded _and_ there was data. This PR updates the logic to only check that the data is no longer loading.

https://user-images.githubusercontent.com/5226211/179022451-66a9c6b7-8f39-456f-9b7e-ab5bd99b09c1.mp4
